deps/dicer/lib/Dicer.js

```diff
@@ -64,15 +63,11 @@ Dicer.prototype.emit = function(ev) {
           self._part.emit('error', new Error(type + ' terminated early due to unexpected end of multipart data'));
           self._part.push(null);
           process.nextTick(function() {
-            self._realFinish = true;
-            self.emit('finish');
-            self._realFinish = false;
+            WritableStream.prototype.emit.call(self, 'finish');
```

Comment:
Can you explain this change? What is the advantage over calling self.emit? Any why we don't need _realFinish?

Reply:
From the original PR by @RolandHeinze

> Therefore, I wrote this PR. It
> 
> * removes the mentioned bug;
> * removes a bug in the HeaderParser constructor functions which computed a wrong value for the variable `end`;
> * removes some unnecessary code in `HeaderParser.prototype._parseHeader()`;
> * removes the variable `_realFinish` and all of it's occurances in `Dicer.js` as it is not necessary, increases the code size, and adds unwanted complexity;
> * removes an unnecessary `else if`clause in the function `Dicer.prototype._oninfo()`;

Reply:
If I do self.emit("finish") atleast one test fails:

```
  dicer-endfinish
    ✔ should properly handle finish
    1) should properly handle finish


  13 passing (17ms)
  1 failing

  1) dicer-endfinish
       should properly handle finish:
     Uncaught AssertionError: Failed to finish
      at Immediate.afterWrite (test/dicer-endfinish.spec.js:48:7)
      at processImmediate (node:internal/timers:464:21)

```

Reply:
1) I don't see why `WritableStream.prototype.emit.call(self, 'finish');` is necessary, given that `self` inherits from `WritableStream` already. Does the fix no longer work if this part is not changed? (later we should rewrite the code to be proper classes either way, but I assume it is not critical for bug at hand);
2) I have my strong doubts than `realFinish` is unnecessary, it doesn't seem to be replaced with any other solution at all. Unless this is essential to fix at hand, I would not do the change as I don't understand original reasoning enough. I'll dig into commit history of original dicer and try to see where and why it was added.

Generally speaking I would prefer to cut down amount of changes to bare minimum necessary to address the bug, and leave non-meaningful refactorings out of it. Considering rather limited code coverage and rather high impact of the library, I'd err on the side of not potentially breaking some very special edge cases we don't know about.

Reply:
reverted, does not impact on the bugfix.

Reply:
@RolandHeinze Can I ask you to explain your original fix? Why changing to `WritableStream.prototype.emit.call(self, 'finish');` is important here?